clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/validators/StandardKubernetesAttributeValidator.groovy

```diff
@@ -26,6 +26,7 @@ class StandardKubernetesAttributeValidator {
   static final prefixPattern = /^[a-z0-9]+$/
   static final cpuPattern = /^(\d+m)?$/
   static final memoryPattern = /^(\d+(Mi|Gi))?$/
+  static final serverGroupNamePattern = /^([a-zA-Z0-9]*-[a-zA-Z0-9]*){2,3}$/
```

Comment:
This would validate `---`, which is not a valid name. You should probably check that a `v\d{3}` exists at the end of the string too.